### PR TITLE
Update to support Django 4.1

### PR DESCRIPTION
**Pull request by chander**
Django 4.1 has changed the name of _sanitize_token to _check_token_format - handle the case(s) where the new name is required to be imported, rather than the old one.

This is identified in ticket #1640

### CODE REVIEW

**Comment by georgedorn** [comment]
I fixed the failing flake8 checks in the master branch; if you rebase off that, the test matrix will re-run.

**Comment by chander** [comment]
Thanks. I'm not sure if you want to add a set of tests for Django 4.1 as well?  I didn't go that far...

**Comment by georgedorn** [comment]
Yeah, it'd be good to at least add django 4.1 to the test matrix.  It should be a fairly straightforward edit of `tox.ini` and `.github/workflows/python-package.yml`.

**Comment by chander** [comment]
I suspect this is mostly there - looks like there's a new docs failure - thoughts?

**Comment by georgedorn** [comment]
The docs error looks like it is related to this change in Django 4.1: https://github.com/django/django/commit/dc9deea8e85641695e489e43ed5d5638134c15c7#diff-e77dfa6935de3fac94356ffd1029a679f8aa23f2979fe9b31aea7f93e1dccad3R568

The cookbook.rst is a bunch of examples, and the tests run those examples as a bare-minimum smoke test to ensure the docs are still up-to-date.

It can be a little tricky to run everything the CI pipeline runs, but I was able to recreate this one locally with:
```
pip install sphinx
sphinx-build -W -b doctest -d /tmp/doctrees/ docs /tmp/html
```

If you have tox and pyenv it is a little easier to mimic what's going on in CI: `tox -e py3.10-docs` but that requires a bit more setup.

There's a few other 4.1 compatibility problems after this one - I can see a couple that spring up when I use the `run_all_tests.sh` approach.  Feel free to keep going if you want, but don't feel obligated - your fixes thus far will get used (with credit) whether or not you carry on through the lengthy process of fixing one issue only to uncover the next for an unpredictable number of iterations.

**Comment by chander** [comment]
Looks better now.  There were a few more changes in there (notable bits to look at were these.  Flake complained and PyCharm is bit overzealous with it's reformatting, sorry:

* https://github.com/django-tastypie/django-tastypie/pull/1642/files#diff-ef132c19dda39d5989a28a3b45124fe7c2df37dafa421fcf0d502edb4d8f80b4R347 

* https://github.com/django-tastypie/django-tastypie/pull/1642/files#diff-ef132c19dda39d5989a28a3b45124fe7c2df37dafa421fcf0d502edb4d8f80b4R347
* https://github.com/django-tastypie/django-tastypie/pull/1642/files#diff-ef132c19dda39d5989a28a3b45124fe7c2df37dafa421fcf0d502edb4d8f80b4R12
* https://github.com/django-tastypie/django-tastypie/pull/1642/files#diff-ef132c19dda39d5989a28a3b45124fe7c2df37dafa421fcf0d502edb4d8f80b4R322

File: tastypie/authentication.py
Comment (georgedorn) [inline tastypie/authentication.py:11]: We try to put this sort of compatibility stuff into `compat.py` to avoid cluttering the other modules with too much of it.  It also provides opportunity to document why, and re-use the same fixes in multiple places.  The token sanitization stuff is a recurring issue so there's already code in `compat.py` to work with it.

**Review by jayvdb — APPROVED** [review]
the important chunk of this has been added to https://build.opensuse.org/package/show/devel:languages:python:django/python-django-tastypie to keep it building on Tumbleweed.

**Comment by chander** [comment]
I was going to move some of this into compat.py ..
will do that shortly.

**Comment by chander** [comment]
BTW, I have a Dockerfile, docker compose file, and some other stuff so that I can run the tests (though they don't run entirely because I suspect I'm missing some dependencies.  Not sure if it's worthwhile to commit that stuff - it's super simple/small

**Comment by georgedorn** [comment]
I'm happy with the patch as it stands; we can ignore the coveralls drop because it is almost certainly inaccurate.  I'll merge this if you're done with fixes, just let me know.  Also, thanks again.

**Comment by chander** [comment]
> I'm happy with the patch as it stands; we can ignore the coveralls drop because it is almost certainly inaccurate. I'll merge this if you're done with fixes, just let me know. Also, thanks again.

I am done with fixes, thanks :-)   Good to merge!

**Comment by chander** [comment]
Any ideas as to when a new release might get cut for this? thanks

**Comment by zell2002** [comment]
bump @chander  @georgedorn  does anyone know when this will get released? Pypi is still showing 4.0

**Comment by bendemboski** [comment]
@georgedorn can you cut a new release? This is blocking me upgrading to Django 4.1

**Comment by teewuane** [comment]
It looks like test coverage dropped by 0.01% and that is blocking this from releasing. I'll take those chances. Please release :)

**Comment by zell2002** [comment]
Does anyone know if this will ever get resolved and released? 
If not that's fine, but we need to know if this repo needs to be forked and continued, or if other members can join and help?

**Comment by srus** [comment]
@georgedorn thank you very much for the release!

**Pull request merged**
